cantor-archive/src/main/java/com/salesforce/cantor/archive/AbstractBaseArchiver.java

```diff
@@ -42,10 +48,10 @@ static ArchiveInputStream getArchiveInputStream(final Path archiveFile) throws I
         return new TarArchiveInputStream(new GzipCompressorInputStream(new BufferedInputStream(Files.newInputStream(archiveFile))));
     }
 
-    static void checkArchiveArguments(final Object instance, final String namespace, final Path destination) {
+    static void checkArchiveArguments(final Object instance, final String namespace, final Path destination) throws IOException {
```

Comment:
This should only throw `IllegalArgumentException`, even if calls to `Files` throw `IOException`.

Reply:
Yeah good point. I'll actually use the standard file length method which doesn't cause an `IOException`.